bftengine/src/bcstatetransfer/BCStateTran.cpp

```diff
@@ -932,7 +932,7 @@ void BCStateTran::sendAskForCheckpointSummariesMsg() {
   lastTimeSentAskForCheckpointSummariesMsg = getMonotonicTimeMilli();
   lastMsgSeqNum_ = uniqueMsgSeqNum();
   metrics_.last_msg_seq_num_.Get().Set(lastMsgSeqNum_);
-  SCOPED_MDC_SEQ_NUM(std::to_string(lastMsgSeqNum_));
+  SCOPED_MDC_SEQ_NUM(std::to_string(config_.myReplicaId) + "-" + std::to_string(lastMsgSeqNum_) + "-0" + "-0");
```

Comment:
If you introduce a new format then this should be part of the macro itself which then takes those variables as arguments.
IIRC, replica ID should be part of the log message already and hence this would be duplication.

Reply:
State Transfer is a separate protocol, we are piggybacking on the bft sequence number positional variable in log format and put there ST "sequence number" which may differ for different parts of protocol - somewhere it receives 2 arguments, somewhere 4. In one plase replica id is my replica id, in the other - id of a peer replica.
So it makes no sense to introduce several new MDC macros for all these cases.

Reply:
The format is simple enough and I wouldn't even notice if it was just one or two occurrences. However, the format (as simple as it is) is duplicated in too many places. If anyone would want to write a parser then what is the source of truth for the parser? Line 935, or 1268, ... ? My point is that this is not a maintainable approach. Humans make mistakes and if we adopt a new line then we might get the format wrong by accident. Or what if we have to change the format for yet unknown reasons?
I know that those are edge case failure scenario. However, what is the harm in building a scalable/maintainable solution from the start?

Reply:
@robem , what's your exact suggestion?

Reply:
There are multiple solutions. Above, I mentioned to use a macro, however, you can also replace the string concatenation with a function that generates the string for you. I'd prefer an inline function over a macro but I'm fine with either.